homeassistant/components/application_credentials/__init__.py

```diff
@@ -88,7 +88,7 @@ async def _process_create_data(self, data: dict[str, str]) -> dict[str, str]:
         domain = result[CONF_DOMAIN]
         if not await _get_platform(self.hass, domain):
             raise ValueError(f"No application_credentials platform for {domain}")
-        return result
+        return result  # type: ignore[no-any-return]
```

Comment:
Can't we cast the result to `dict[str, str]` if we know that's what it is?

Reply:
Yeah, that would be possible. However, I usually try to avoid `cast` as much as possible. Especially if it's a trivial one only for the return type.

I explained the different options around `no-any-return` here in more detail: https://github.com/home-assistant/core/pull/106907#discussion_r1439789390
What it comes down to basically is that there is ever a typing solution for `voluptuous`, this would be typed automatically in which case we would just need to remove the `type: ignore`.

Reply:
Ok.